app/controllers/openid_application_controller.rb

```diff
@@ -54,18 +54,30 @@ def authenticate_token
       return true
     end
 
+    return false if @session.uuid.nil?
+
     @current_user = OpenidUser.find(@session.uuid)
   end
 
   def populate_payload_for_launch_scope
-    launch = fetch_smart_launch_context
-    token.payload[:launch] = base64_json?(launch) ? JSON.parse(Base64.decode64(launch)) : { patient: launch }
+    if @session.launch.nil?
```

Comment:
This is going to break client_credential tokens with launch in the payload because the session never gets instantiated in the first place.

Reply:
Here's how to get the breaking token
```sh
docker run \
  vasdvp/lighthouse-auth-utils:latest auth-cc \
  --client-id="$CC_CLIENT_ID" \
  --client-secret="$CC_CLIENT_SECRET" \
  --authorization-url="$SANDBOX_AUTHORIZATION_URL" \
  --scope="launch" \
  --launch="123V456" | jq .access_token | tr -d '\"'
```

Reply:
👍 added cc session builder for launch scope to match that of launch/patient scope.